docs/CONTRIBUTING.md

```diff
@@ -141,59 +162,107 @@ task test-integration
+If you don't see any error, hit http://127.0.0.1:8000 with your browser to
+navigate the generated docs.
+
+### Docs publishing
+
+The present git repository has a special branch called `gh-pages` that contains
+the generated HTML code for the docs website; every time a change is pushed to
+this special branch, GitHub automatically triggers a [deployment][8] to pull the
+change and publish a new version of the website. Do not open Pull Requests to
+push changes to the `gh-pages` branch, that will be done exclusively from the
+CI.
+
+### Docs versioning
+
+In order to provide support for multiple Arduino CLI releases, Documentation is
+versioned so that visitors can select which version of the documentation website
+should be displayed. Unfortunately this feature isn't provided by neither GitHub
+pages nor MkDocs, so we had to implement it on top of the generation process.
+
+Before delving into the details of the generation process, here follow some
+requirements that were established to provide versioned documentation:
+
+* A special version of the documentation called `dev` is provided to reflect the
+  status of the Arduino CLI on the `master` branch - this includes unreleased
+  features and bugfixes.
+* Docs are versioned after the minor version of an Arduino CLI release. For
+  example, Arduino CLI `0.99.1` and `0.99.2` will be both covered by
+  documentation version `0.99`.
+* The landing page of the documentation website will automatically redirect
+  visitors to the most recently released version of the Arduino CLI.
+
+To implement the requirements above, the execution of MkDocs is wrapped using a
+CLI tool called [Mike][10] that does few things for us:
+
+* It runs MkDocs targeting subfolders named after the Arduino CLI version, e.g.
+  documentation for version `0.10.1` can be found under the folder `0.10`.
+* It injects an HTML control into the documentation website that let visitors
+  choose which version of the docs to browse from a dropdown list.
+* It provides a redirect to a version we decide when visitors hit the landing
+  page of the documentation website.
+* It pushes generated contents to the `gh-pages` branch.
+
+> **Note:** unless you're working on the generation process itself, you should
+> never run Mike from a local environment, either directly or through the Task
+> `docs:publish`. This might result in unwanted changes to the public website.
+
+### Docs automation
+
+In order to avoid unwanted changes to the public website hosting the Arduino
+CLI documentation, only Mike is allowed to push changes to the `gh-pages` branch,
+and this only heppens from within the CI, in a workflow named [docs][11].
+
+The CI is responsible for guessing which version of the Arduino CLI we're
+building docs for, so that generated contents will be stored in the appropriate
+section of the documentation website. This guessing might be fairly complex,
+reason why the logic is implemented in a Python script called [`build.py`][12].
```

Comment:
```suggestion
section of the documentation website. Because this guessing might be fairly
complex, the logic is implemented in a Python script called [`build.py`][12].
```